src/include/s390_aes.h

```diff
@@ -308,12 +312,15 @@ static inline int s390_aes_ecb_sw(unsigned int function_code,
 	unsigned int direction;
 	unsigned int key_size = (function_code & 0x0f) *
 				sizeof(ica_aes_key_single_t);
+	int rc = 0;
 
 #ifdef ICA_FIPS
-	if ((fips & ICA_FIPS_MODE) && (!FIPS_mode()))
+	if ((fips & ICA_FIPS_MODE) && (!openssl_in_fips_mode()))
```

Comment:
For OpenSSL 3.0 this simplifies to `(fips & ICA_FIPS_MODE) && (!fips)` which is always false.